File: src/main/java/sk/hudak/prco/parser/HtmlParser.java
package sk.hudak.prco.parser;

import sk.hudak.prco.api.EshopUuid;
import sk.hudak.prco.dto.internal.ProductNewData;
import sk.hudak.prco.dto.internal.ProductUpdateData;

import java.util.List;

public interface HtmlParser {

    List<String> searchProductUrls(EshopUuid eshopUuid, String searchKeyWord);

    /**
     * Urobi connet na danu URL a vyparsuje html data o produkte.
     *
     * @param productUrl
     * @return
     */
    ProductNewData parseProductNewData(String productUrl);

    ProductUpdateData parseProductUpdateData(String productUrl);


}

File: src/main/java/sk/hudak/prco/parser/HtmlParser.kt
package sk.hudak.prco.parser

import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.dto.ProductNewData
import sk.hudak.prco.dto.ProductUpdateData

interface HtmlParser {

    fun searchProductUrls(eshopUuid: EshopUuid, searchKeyWord: String): List<String>

    /**
     * Urobi connect na danu URL a vyparsuje html data o produkte.
     *
     * @param productUrl
     * @return
     */
    fun parseProductNewData(productUrl: String): ProductNewData

    fun parseProductUpdateData(productUrl: String): ProductUpdateData
}